transport/conn.go

```diff
@@ -639,8 +652,13 @@ func MakeResponseHandlerWithError(err error) ResponseHandler {
 	return h
 }
 
-func (c *Conn) sendRequest(req frame.Request, compress, tracing bool) (frame.Response, error) {
-	c.sendController()
+func (c *Conn) sendRequest(ctx context.Context, req frame.Request, compress, tracing bool) (frame.Response, error) {
+	if err := c.sendController(ctx); err != nil {
+		return nil, err
+	}
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
```

Comment:
shouldn't sendController return error when context is cancelled? If so, this check is redundant

Reply:
This check is not redundant as sendController could finish immediately without checking the ctx,
however it could return ctx.Err() instead of nil, making this check redundant (making it easier not to forget)